### PR TITLE
Changes to make xentropysoftmax load/store vectorized when possible:

### PR DESCRIPTION
**Pull request by FDecaYed**
Increase default ILP so that each thread handle 16 Bytes data in one step
Make thread load/store longest vector possible
Support shift for blocks that are not aligned. Greatly improve performance for cases where `classes % 8 > 0`

### CODE REVIEW

File: apex/contrib/csrc/xentropy/xentropy_kernel.cu [hunk 1]
```diff
@@ -123,7 +124,7 @@ const int max_threads = 1024;
 inline dim3 SoftMax_getBlockSize(int ILP, uint64_t dim_size) {
   uint64_t block_size = 1;
   uint64_t max_block_size = std::min(dim_size / ILP, static_cast<uint64_t>(max_threads));
-  while (block_size < max_block_size) block_size *= 2;
+  while (block_size < (max_block_size/2)) block_size *= 2;
```
Comment (slayton58) [inline apex/contrib/csrc/xentropy/xentropy_kernel.cu:127]: Why is this change needed? Should it be `max_block_size / ILP`?

Comment (FDecaYed) [reply apex/contrib/csrc/xentropy/xentropy_kernel.cu:127]: resolved with simon:
`/n` here means `(n-1)/n` portion of the block will be vectorized in case of shifting due to not multiple/aligned. Too high of `n` makes block size too small, so I chose 2 here.
